lib/floe/workflow/intrinsic_function.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+module Floe
+  class Workflow
+    class IntrinsicFunction
+      class << self
+        def klass(payload)
+          function_module, function =
+            payload.match(/^(?<module>\w+)\.(?<function>\w+)\(.*\)$/)
+                   .named_captures.values_at("module", "function")
+
+          case "#{function_module}.#{function}"
+          when "States.StringToJson"
+            Floe::Workflow::IntrinsicFunctions::States::StringToJson
+          else
+            raise NotImplementedError
+          end
+        end
+
+        def value(payload, context, input = {})
+          new(payload).value(context, input)
+        end
+
+        private alias_method :orig_new, :new
+
+        def new(*args)
+          if self == Floe::Workflow::IntrinsicFunction
+            klass(*args).new(*args)
+          else
+            orig_new(*args)
+          end
+        end
+      end
+
+      attr_reader :args
+
+      def initialize(payload)
+        args  = payload.match(/^\w+\.\w+\((.*)\)$/).captures.first
```

Comment:
Minor, but we might want to move this regex out to a constant and reuse on line 42

Reply:
Actually, is this the same regex as line 9 as well?

Reply:
Done